modules/features/sharing/src/main/kotlin/au/com/shiftyjelly/pocketcasts/sharing/ui/BackgroundAssetController.kt

```diff
@@ -0,0 +1,100 @@
+package au.com.shiftyjelly.pocketcasts.sharing.ui
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.graphics.LinearGradient
+import android.graphics.Paint
+import android.graphics.Shader
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.ExperimentalComposeApi
+import androidx.compose.ui.graphics.asAndroidBitmap
+import androidx.compose.ui.graphics.toArgb
+import androidx.core.graphics.applyCanvas
+import androidx.core.graphics.scale
+import au.com.shiftyjelly.pocketcasts.sharing.ui.CardType.Horiozntal
+import au.com.shiftyjelly.pocketcasts.sharing.ui.CardType.Square
+import au.com.shiftyjelly.pocketcasts.sharing.ui.CardType.Vertical
+import dev.shreyaspatil.capturable.controller.CaptureController
+import dev.shreyaspatil.capturable.controller.rememberCaptureController
+import java.io.File
+import kotlin.math.roundToInt
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+
+internal interface BackgroundAssetController {
+    fun captureController(type: CardType): CaptureController
+
+    suspend fun capture(type: CardType): Result<File>
+
+    companion object {
+        private const val BITMAP_WIDTH = 1080
+        private const val BITMAP_HEIGHT = 1920
```

Comment:
The code seems to be scaling the height of the image for the horizontal card but it's coming through in Instagram at the same height. Is that the normal behaviour on Instagram?

<img  width="300" src="https://github.com/user-attachments/assets/faf14bdc-32a6-4073-997b-0258101da4a7" />

Can I confirm this isn't the final design? The colors and text sizes don't seem quite right.

Reply:
> The code seems to be scaling the height of the image for the horizontal card but it's coming through in Instagram at the same height. Is that the normal behaviour on Instagram?

I'm not sure what do you mean. How would you expect horizontal card too look like? It needs to be scaled uniformly.

> Can I confirm this isn't the final design? The colors and text sizes don't seem quite right.

Colors are still being designed: pdeCcb-50w-p2#comment-4887

Font size's are correct. From top to bottom they are `H70`, `H40`, and `H70`. HR2vcQrWcS0scsolKZBitg-fi-2336_11067